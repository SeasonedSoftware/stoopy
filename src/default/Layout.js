import React from "react";
import {
  CircularProgress,
  FormControl,
  Box,
  FormControlLabel,
  RadioGroup,
  FormLabel,
  Radio,
  MenuItem,
  Button,
  Typography,
  Select,
  InputLabel,
  Container
} from "@material-ui/core";

// LAYOUT COMPONENTS

// Button to next step. This should come from props, and default to basic one
// if none was provided

const justifyContent = ({ children, maxWidth = "sm", ...props }) => (
  <Container maxWidth={maxWidth}>
    <Box mb={2} {...props}>
      {children}
    </Box>
  </Container>
);

export const NextButton = props => {
  return (
    <div>
      <Button type="submit" variant="contained" color="primary" {...props}>
        Next
      </Button>
    </div>
  );
};

export const BackButton = props => {
  return (
    <div>
      <Button type="button" variant="contained" color="secondary" {...props}>
        Back
      </Button>
    </div>
  );
};

export const ProgressTracker = ({ progress }) => {
  return (
    <div style={{ marginTop: 10 }}>
      <Typography variant="h5">
        {progress.currentStep}/{progress.totalSteps}
      </Typography>
    </div>
  );
};
// Form header, title, etc...
export const FormHeader = ({
  title = "Almost there! Just a few more steps..."
}) => {
  return (
    <Typography style={{ marginBottom: 20 }} variant="h5">
      {title}
    </Typography>
  );
};

// Default Loading component for transitions. Not sure if will still
// be used after setting up transitions
export const Loading = props => (
  <div textAlign="center" maxWidth="lg" {...props}>
    <CircularProgress />
  </div>
);
