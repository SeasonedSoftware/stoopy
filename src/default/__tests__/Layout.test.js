import React from 'react'
import renderer from 'react-test-renderer'
import { NextButton } from '../Layout'

it('renders correctly', () => {
  const props = { foo: 'bar' }
  const tree = renderer.create(<NextButton {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
