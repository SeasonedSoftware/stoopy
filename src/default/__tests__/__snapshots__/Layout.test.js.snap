// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    foo="bar"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Next
    </span>
  </button>
</div>
`;
